node/node.go

```diff
@@ -1225,8 +1225,8 @@ func (node *AlgorandFullNode) SetCatchpointCatchupMode(catchpointCatchupMode boo
 		if catchpointCatchupMode {
 			// stop..
 			defer func() {
-				node.mu.Unlock()
 				node.waitMonitoringRoutines()
```

Comment:
Is it not possible to deadlock here waiting for `oldKeyDeletionThread` to return, since we have `node.mu`? https://github.com/algorand/go-algorand/blob/master/node/node.go#L1123

This comment https://github.com/algorand/go-algorand/blob/master/node/node.go#L404 makes me concerned about that possibility.

Reply:
you are right, waiting with lock taken is bad. Ok, not as easy as I thought.